packages/flutter/lib/src/widgets/page_transitions_builder.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport 'package:flutter/material.dart';
+library;
+
+import 'basic.dart';
+import 'framework.dart';
+import 'pages.dart';
+import 'transitions.dart';
+
+/// Defines a page transition animation for a [PageRoute].
+///
+/// PageTransitionsBuilder can be used directly with widget layer primitives
+/// in any design system. Custom [PageRoute] subclasses can accept a
+/// PageTransitionsBuilder and delegate to its [buildTransitions] method when
+/// overriding [ModalRoute.buildTransitions]. This enables reusable transition
+/// animations that work with [Navigator] and other navigation primitives.
+///
+/// ## Example usage
+///
+/// {@tool dartpad}
+/// This example shows how to create a custom [PageTransitionsBuilder] that
+/// slides the new page in from the right while fading it in, and how to use
+/// it with a custom [PageRoute].
+///
+/// ** See code in examples/api/lib/widgets/page_transitions_builder/page_transitions_builder.0.dart **
+/// {@end-tool}
+///
+/// This class is used by [PageTransitionsTheme] to define a [MaterialPageRoute]
```

Comment:
Turbo nit: add to the beginning of this, "As an example of usage, in the Material library this class is used by...."